tfjs-converter/src/executor/model_analysis.ts

```diff
@@ -179,50 +202,57 @@ export function getNodesInTopologicalOrder(
   }
 
   // Filter out unreachable nodes and build the ordered node list.
-  const filteredOrderedNodes =
-      orderedNodeNames.filter((name) => predefinedReachableNodeNames.has(name))
-          .map((name) => nameToNode.get(name)!);
+  const filteredOrderedNodes = orderedNodes.filter(
+      (node) => predefinedReachableNodeNames.has(node.name));
 
-  // Validates node orders
-  // TODO: Turn validation on/off with tf env flag.
-  {
-    const nodeNameToOrder = new Map<string, number>(
-        filteredOrderedNodes.map((node, order) => [node.name, order]));
-    const predefinedNodeNames =
-        new Set(predefinedNodes.map((node) => node.name));
-    const isPredefined = (node: Node|string) =>
-        predefinedNodeNames.has(typeof node === 'string' ? node : node.name);
-    const willBeExecutedNodeNames =
-        new Set(filteredOrderedNodes.map((node) => node.name));
-    const willBeExecuted = (node: Node|string) => willBeExecutedNodeNames.has(
-        typeof node === 'string' ? node : node.name);
+  return filteredOrderedNodes;
+}
 
-    for (const node of filteredOrderedNodes) {
-      for (const child of node.children.filter(willBeExecuted)) {
-        if (!nodeNameToOrder.has(child.name)) {
-          throw new Error('TopologicalSortError: Child is unreachable.');
-        }
-        if (nodeNameToOrder.get(node.name) > nodeNameToOrder.get(child.name)) {
-          throw new Error(
-              'TopologicalSortError: Node has greater order than its child.');
-        }
+/**
+ * This is a helper function of `getNodesInTopologicalOrder`.
+ * Validates property: given nodes `a` and `b`, Order(a) > Order(b) if `a`
+ * is a child of `b`. This function throws an error if validation fails.
+ *
+ * @param orderedNodes Graph nodes in execution order.
+ * @param predefinedNodes Graph inputs, weights, and init nodes. Nodes in this
+ *     list must have distinct names.
+ */
+function validateNodesExecutionOrder(
+    orderedNodes: Node[], predefinedNodes: Node[]) {
+  const nodeNameToOrder = new Map<string, number>(
+      orderedNodes.map((node, order) => [node.name, order]));
+  const predefinedNodeNames = new Set(predefinedNodes.map((node) => node.name));
+  const isPredefined = (node: Node|string) =>
+      predefinedNodeNames.has(typeof node === 'string' ? node : node.name);
+  const willBeExecutedNodeNames =
+      new Set(orderedNodes.map((node) => node.name));
+  const willBeExecuted = (node: Node|string) =>
+      willBeExecutedNodeNames.has(typeof node === 'string' ? node : node.name);
+
+  for (const node of orderedNodes) {
+    for (const child of node.children.filter(willBeExecuted)) {
+      if (!nodeNameToOrder.has(child.name)) {
+        throw new Error(`TopologicalSortError: Child ${child.name} of node ${
```

Comment:
Tiny nit: We could have a custom error type

```typescript
class TopologicalSortError extends Error {
  constructor(message: string) {
    super(`TopologicalSortError: ${message}`);
  }
}
```

...but this is not necessary and should not block the PR.